### DISCUSSION TITLE
Global project configuration

### DISCUSSION THREAD

**Discussion by S3gmentati0nFault** [typst/typst#3295] [Q&A] [answered]
Hi!

As many others I am coming here from a Latex experience and I am now comfortable with my .sty setup that I basically use for every project, it's very easy to use and it allows me to keep all of the macros in one place, without the need to jump from one file to the other.

I was wondering if there was the possibility to do this in Typst as well (there was a previous thread similar to this one <a href="https://github.com/typst/typst/discussions/2201">here</a>

Disclamer: consider that I am pretty new to the typesetting language and I have just a general idea of how the language is structured, I tried looking for information on the official website but I couldn't find anything. If there is a resource dedicated to this just let me know!

**Comment by frozolotl** [comment, marked as answer]
That is totally possible in Typst.

You can check out [Making a Template](https://typst.app/docs/tutorial/making-a-template/) for a small tutorial on how templates are created.

Instead of copying the file around, I recommend creating a local package that can be imported by every project: https://github.com/typst/packages?tab=readme-ov-file#local-packages

**Reply by S3gmentati0nFault** [reply]
Thank you very much for the insight, I will take some time in the next couple of days to study the solution!

**Reply by kapple19** [reply]
What if we need our templates to be available in a repository along with documents that all use these templates? We'd need to maintain two copies of each template which is a very poor user experience.

**Reply by PgBiel** [reply]
You can have a symlink from your local packages folder to your repository in that case, and have the documents import from the symlink path with `#import "/template.typ"`. Typst won't make a distinction between a symlink and a regular file here.

**Reply by kapple19** [reply]
Oh awesome, I'll give it a shot.
